vars/sonarExecute.groovy

```diff
@@ -0,0 +1,86 @@
+import com.sap.piper.ConfigurationHelper
+import com.sap.piper.Utils
+
+import groovy.transform.Field
+import groovy.text.SimpleTemplateEngine
+
+@Field String STEP_NAME = 'sonarExecute'
+@Field Set STEP_CONFIG_KEYS = [
+    'dockerImage', // the image to run the sonar-scanner
+    'instance', // the instance name of the Sonar server configured in the Jenkins
+    'options',
+    'projectVersion',
+    // needed for voter
+    'disableInlineComments', // set to true to only enable a summary comment on the pull-request
+    'isVoter', // enables the preview mode
+    'changeId', // the pull-request number
+    'githubApiUrl', // URL to access GitHub WS API | default: https://api.github.com
+    'githubOrg',
+    'githubRepo',
+    'githubTokenCredentialsId'
+]
+@Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS
+@Field Set GENERAL_CONFIG_KEYS = STEP_CONFIG_KEYS
+
+def call(Map parameters = [:], Closure body = null) {
+    handlePipelineStepErrors(stepName: STEP_NAME, stepParameters: parameters) {
+        def utils = parameters.juStabUtils ?: new Utils()
+        def script = parameters.script ?: [commonPipelineEnvironment: commonPipelineEnvironment]
+        // load default & individual configuration
+        Map config = ConfigurationHelper
+            .loadStepDefaults(this)
+            .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS)
+            .mixinStageConfig(script.commonPipelineEnvironment, parameters.stageName?:env.STAGE_NAME, GENERAL_CONFIG_KEYS)
+            .mixinStepConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS)
+            .mixin(
+                projectVersion: script.commonPipelineEnvironment.getArtifactVersion()?.tokenize('.')?.get(0),
+                githubOrg: script.commonPipelineEnvironment.getGithubOrg(),
+                githubRepo: script.commonPipelineEnvironment.getGithubRepo(),
+                changeId: env.CHANGE_ID
+            )
+            .mixin(parameters, PARAMETER_KEYS)
+            .use()
+        // check mandatory parameters
+        if(config.isVoter){
+            new ConfigurationHelper(config)
```

Comment:
[minor] No need for a new instance of ConfigurationHelper. The old one from above can be used (we need a variable in this case, of course, in order to hold the reference).

Reply:
will vanish with implementation of #237